main.tf

```
@@ -24,14 +24,14 @@ resource "aws_db_instance" "default" {
 
   identifier            = module.this.id
   name                  = var.database_name
-  username              = var.database_user
-  password              = var.database_password
+  username              = length(var.replicate_source_db) > 0 ? null : var.database_user
```

Lets set the logic so the falsey portion returns null. Please do this for all of the new logic to keep it consistent

```suggestion
  username              = length(var.replicate_source_db) == 0 ? var.database_user : null
```

---

@mihaiplesa please do this for all of the respective changes, not just this one. 😄

---

thanks, let me know how it looks now :D